src/main/webapp/js/common/react/components/DataFetcher.js

```diff
@@ -0,0 +1,71 @@
+import checkTesting from '../../../checkTesting.js';
+import {basicErrorHandling} from '../../../fetch_handler.js';
+checkTesting();
+
+/**
+ * A generic data fetching component that can be used for fetching data for any
+ * component.
```

Comment:
For such a generic component, I think it doesn't hurt to point to or provide a usage example in the comments.

Reply:
Need to describe "What does this do for me" and "How do I use it"

Reply:
That's a good point, as I definitely intend to this to be used by more people than just me. I'll improve the documentation for this.

Reply:
Yep! Sometimes, the more general your code, the higher the comment/code ratio :) I think I remember we looked at some examples in the Google codebase where there's about 3 lines of code and 100s of comments to explain all the usage, implications, etc.

Reply:
That makes sense -- the more general the code the more people are likely to use it. And yeah I definitely remember looking at code like that haha.